File: cpp/0073.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        int m = matrix.size();
        int n = matrix[0].size();
        vector<bool> row(m, false);
        vector<bool> col(n, false);

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (matrix[i][j] == 0) {
                    row[i] = true;
                    col[j] = true;
                }
            }
        }
        for (int i = 0; i < m; i++) {
            if (row[i] == false) {
                continue;
            }
            for (int j = 0; j < n; j++) {
                matrix[i][j] = 0;
            }
        }
        for (int i = 0; i < n; i++) {
            if (col[i] == false) {
                continue;
            }
            for (int j = 0; j < m; j++) {
                matrix[j][i] = 0;
            }
        }

        return;
    }
};

File: 0073.py
class Solution(object):
    def setZeroes(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: void Do not return anything, modify matrix in-place instead.
        """
        # backup
        temp = [row[:] for row in matrix]
        
        # variables
        m = len(matrix)
        n = len(matrix[0])
        
        for i in range(0, m):
            for j in range(0, n):
                if temp[i][j] == 0:
                    # replace row
                    matrix[i] = [0] * n
                    # replace col
                    for k in range(0, m):
                        matrix[k][j] = 0